src/scripts/site.js

```diff
@@ -287,103 +413,191 @@ var Dashboard = {
-        var list = [];
-        window.fetch || list.push("fetch"), "function" != typeof Object.assign && list.push("objectassign"), Array.prototype.filter || list.push("arraypolyfills"), Function.prototype.bind || list.push("functionbind"), window.requestAnimationFrame || list.push("raf"), require(list, function() {
-            createConnectionManager().then(function() {
-                console.log("initAfterDependencies promises resolved"), require(["globalize", "browser"], function(globalize, browser) {
-                    window.Globalize = globalize, loadCoreDictionary(globalize).then(function() {
-                        onGlobalizeInit(browser)
-                    })
-                })
-            })
-        })
+        if ("android" === self.appMode) {
+            define("nativedirectorychooser", ["cordova/nativedirectorychooser"], returnFirstDependency);
+        }
+
+        define("livetvcss", ["css!css/livetv.css"], returnFirstDependency);
+        define("detailtablecss", ["css!css/detailtable.css"], returnFirstDependency);
+        define("buttonenabled", ["legacy/buttonenabled"], returnFirstDependency);
+        var promises = [];
+
+        if (!window.fetch) {
+            promises.push(require(["fetch"]));
+        }
+
+        if ("function" != typeof Object.assign) {
+            promises.push(require(["objectassign"]));
+        }
+
+        if (!Array.prototype.filter) {
+            promises.push(require(["arraypolyfills"]));
+        }
+
+        if (!Function.prototype.bind) {
+            promises.push(require(["functionbind"]));
+        }
+
+        if (!window.requestAnimationFrame) {
+            promises.push(require(["raf"]));
+        }
+
+        Promise.all(promises).then(function () {
+            createConnectionManager().then(function () {
+                console.log("initAfterDependencies promises resolved");
+
+                require(["globalize", "browser"], function (globalize, browser) {
+                    window.Globalize = globalize;
+                    loadCoreDictionary(globalize).then(function () {
+                        onGlobalizeInit(browser);
+                    });
+                });
+            });
+        });
     }
 
     function loadCoreDictionary(globalize) {
         var languages = ["ar", "be-by", "bg-bg", "ca", "cs", "da", "de", "el", "en-gb", "en-us", "es", "es-ar", "es-mx", "fa", "fi", "fr", "fr-ca", "gsw", "he", "hi-in", "hr", "hu", "id", "it", "kk", "ko", "lt-lt", "ms", "nb", "nl", "pl", "pt-br", "pt-pt", "ro", "ru", "sk", "sl-si", "sv", "tr", "uk", "vi", "zh-cn", "zh-hk", "zh-tw"];
-        var translations = languages.map(function(i) {
+        var translations = languages.map(function (i__t) {
```

Comment:
Please replace variables like this with more descriptive names. This one should be `language`.

Reply:
done